### PR TITLE
Fix nightly builds

### PR DESCRIPTION
**Pull request by paulfantom**
Fix Jenkis pipeline to enable nightly build on master branch.

This cannot be tested as PR, and will be tested manually via Jenkins Web interface with builds on master branch.

### CODE REVIEW

**Comment by paulfantom** [comment]
This was tested in build https://jenkins-tectonic-installer.prod.coreos.systems/job/tectonic-installer/job/master/895/

**Comment by paulfantom** [comment]
`unstash` fails due to existing file also we aren't cleaning workspace after build, so all files needed to build a container exist and there is no need to unstash.

File: Jenkinsfile [hunk 1]
```diff
@@ -103,7 +103,6 @@ pipeline {
       steps {
         withCredentials(quayCreds) {
           ansiColor('xterm') {
-            unstash 'tectonic-tarball'
```
Comment (enxebre) [inline Jenkinsfile]: should we remove this as well then https://github.com/coreos/tectonic-installer/blob/master/Jenkinsfile#L92?

Comment (paulfantom) [reply Jenkinsfile]: I think that line produces an artifact which can be downloaded later, so I would leve it as is.

**Pull request merged**